dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/user/DatadogUserInfoProviderTest.kt

```diff
@@ -53,36 +53,51 @@ internal class DatadogUserInfoProviderTest {
 
     @Test
     fun `M return saved userInfo W setUserInfo() and getUserInfo()`(
-        @Forgery userInfo: UserInfo
+        @Forgery userInfo: UserInfo,
+        @StringForgery userId: String
     ) {
+        // Given
+        val nonNullUserId = userInfo.id ?: userId
+        val validUserInfo = userInfo.copy(id = nonNullUserId)
+
         // When
-        testedProvider.setUserInfo(userInfo.id, userInfo.name, userInfo.email, userInfo.additionalProperties)
+        testedProvider.setUserInfo(nonNullUserId, userInfo.name, userInfo.email, userInfo.additionalProperties)
         val result = testedProvider.getUserInfo()
 
         // Then
-        assertThat(result).isEqualTo(userInfo)
+        assertThat(result).isEqualTo(validUserInfo)
     }
 
     @Test
-    fun `M delegate to persister W setUserInfo`(@Forgery userInfo: UserInfo) {
+    fun `M delegate to persister W setUserInfo`(
+        @Forgery userInfo: UserInfo,
+        @StringForgery userId: String
+    ) {
+        // Given
+        val nonNullUserId = userInfo.id ?: userId
+        val validUserInfo = userInfo.copy(id = nonNullUserId)
+
         // When
-        testedProvider.setUserInfo(userInfo.id, userInfo.name, userInfo.email, userInfo.additionalProperties)
+        testedProvider.setUserInfo(nonNullUserId, userInfo.name, userInfo.email, userInfo.additionalProperties)
 
         // Then
-        verify(mockWriter).write(userInfo)
+        verify(mockWriter).write(validUserInfo)
     }
 
     @Test
     fun `M keep existing properties W addUserProperties() is called`(
         @Forgery userInfo: UserInfo,
+        @StringForgery userId: String,
         @StringForgery forge: Forge
```

Comment:
minor: strange that there is a `StringForgery` annotation here, it shouldn't have any effect and can be removed.